seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/Catalog.java

```diff
@@ -125,17 +124,14 @@ default Optional<Factory> getFactory() {
     default List<CatalogTable> getTables(ReadonlyConfig config) throws CatalogException {
         // Get the list of specified tables
         List<String> tableNames = config.get(CatalogOptions.TABLE_NAMES);
-        List<CatalogTable> catalogTables = Collections.synchronizedList(new ArrayList<>());
-        if (CollectionUtils.isNotEmpty(tableNames)) {
-            tableNames
-                    .parallelStream()
-                    .forEach(
-                            tableName -> {
-                                TablePath tablePath = TablePath.of(tableName);
-                                if (this.tableExists(tablePath)) {
-                                    catalogTables.add(this.getTable(tablePath));
-                                }
-                            });
+        List<CatalogTable> catalogTables = new ArrayList<>();
+        if (tableNames != null && !tableNames.isEmpty()) {
+            for (String tableName : tableNames) {
+                TablePath tablePath = TablePath.of(tableName);
```

Comment:
It's better to use parallelStream here, in the test, if we have many tables this part might cost several minutes.

Reply:
If you add Parallelstream, it will cause disorder